internal/pkg/source/source.go

```diff
@@ -232,25 +233,49 @@ func sourcesFromBlocks(fn *ssa.Function, conf classifier) []*Source {
 
 		for _, instr := range b.Instrs {
 			switch v := instr.(type) {
-			// Looking for sources of PII allocated within the body of a function.
+			default:
+				continue
+
+			// source variable allocated
 			case *ssa.Alloc:
-				if conf.IsSource(utils.Dereference(v.Type())) && !isProducedBySanitizer(v, conf) {
-					sources = append(sources, New(v, conf))
+				if isProducedBySanitizer(v, conf) {
+					continue
 				}
 
-				// Handling the case where PII may be in a receiver
-				// (ex. func(b *something) { log.Info(something.PII) }
+			// source returned from call
+			case *ssa.Call:
+				if isProducedBySanitizer(v, conf) {
+					continue
+				}
+
+			// source obtained from field
 			case *ssa.FieldAddr:
-				if conf.IsSource(utils.Dereference(v.Type())) {
-					sources = append(sources, New(v, conf))
+
+			// source obtained from array by index
+			case *ssa.IndexAddr:
+
+			// source obtained from map by key
+			case *ssa.Lookup:
+
+			// source received from chan
+			case *ssa.UnOp:
+				// not a <-chan operation
+				if v.Op != token.ARROW {
+					continue
 				}
 			}
+
+			// all of the above instructions are values as per ssa/doc.go
+			v := instr.(ssa.Value)
+			if conf.IsSource(utils.Dereference(v.Type())) {
```

Comment:
Consider the more idiomatic

```golang
if v := instr.(ssa.Value); conf.IsSource(utils.Dereference(v.Type())) {
```

Reply:
Argh. I always miss these opportunities. Good catch, thanks!